gubernator/github_auth.py

```diff
@@ -41,7 +41,7 @@ class Endpoint(view_base.BaseHandler):
     def github_client(self):
         client = self.app.config['github_client']
         if not client:
-            self.abort(500)
+            self.abort(501, body_template='An admin must <a href="/config">configure Github secrets</a> first.')
```

Comment:
Wouldn't a 4xx level error here be more appropriate? I associate 5xx with unexpected transient errors

Reply:
I should probably just use 500.

4xx = you screwed up
5xx = we (the server) screwed up

Reply:
Seems like this is 4xx: you screwed up, you haven't confgured things yet?

Reply:
It's a server misconfiguration, so it's 500.

"you" is the client.

The user clicking that link might not be an admin at all.